src/poetry/core/masonry/builders/wheel.py

```diff
@@ -69,29 +67,31 @@ def make_in(
         poetry: Poetry,
         directory: Path | None = None,
         original: Path | None = None,
-        executable: str | None = None,
+        executable: Path | None = None,
         editable: bool = False,
     ) -> str:
         wb = WheelBuilder(
             poetry,
-            target_dir=directory,
             original=original,
             executable=executable,
             editable=editable,
         )
-        wb.build()
+        wb.build(target_dir=directory)
 
         return wb.wheel_filename
 
     @classmethod
-    def make(cls, poetry: Poetry, executable: str | None = None) -> None:
+    def make(cls, poetry: Poetry, executable: Path | None = None) -> None:
         """Build a wheel in the dist/ directory, and optionally upload it."""
         cls.make_in(poetry, executable=executable)
 
-    def build(self) -> None:
+    def build(
+        self,
+        target_dir: Path | None = None,
+    ) -> Path:
         logger.info("Building wheel")
 
-        dist_dir = self._target_dir
+        dist_dir = target_dir or (self._poetry.file.parent / "dist")
```

Comment:
I'd consider the fallback target directory a property of the base class.

Reply:
can't say I'm too excited either way but have introduced `default_target_dir()`